lib/tlssock.h

```diff
@@ -29,14 +29,40 @@
 #define TLS_CLT_HANDSHAKE 1
 #define TLS_SRV_HANDSHAKE 2
 
+#define TLS_OPT_PEER_SUBJECT_DN 1
+
+#define TLS_CLIENT_CERT_IGNORE 0
+#define TLS_CLIENT_CERT_REQUEST 1
+#define TLS_CLIENT_CERT_REQUIRE 2
+
 typedef struct {
   void *misc;
 
   ssize_t (*psk)(void *misc, char **username, uint8_t **key);
+  struct {
+    int (*getcert)(void *misc,
+                const char **requested_ca_dn,
+                char **cert_uri,
+                char **key_uri,
+                char **pin);
+    const char *cafile;
+    int insecure;
```

Comment:
If I understand the code correctly, `insecure` means that we trust all certificates. I wonder if we should disallow that mode altogether. In that case, `cafile = NULL` means trust the system CAs.